### PR TITLE
ioctl,types: telemetry log update based NVMe v2.1

### PR DESCRIPTION
**Pull request by sc108-lee**
MCDA added for Telemetry Host-Initiated Log creation

Log member variable added
Host-Initiated Log - THS
Controller-Initiated Log - TCS

### CODE REVIEW

File: src/nvme/ioctl.h [hunk 1]
```diff
@@ -1548,7 +1564,7 @@ static inline int nvme_get_log_create_telemetry_host(int fd,
 		.nsid = NVME_NSID_NONE,
 		.csi = NVME_CSI_NVM,
 		.lsi = NVME_LOG_LSI_NONE,
-		.lsp = NVME_LOG_TELEM_HOST_LSP_CREATE,
+		.lsp = (__u8)((mcda << 1) | NVME_LOG_TELEM_HOST_LSP_CREATE),
```
Comment (igaw) [inline src/nvme/ioctl.h:1567]: I assume this is the thing which `5.1.12.1.8 Telemetry Host-Initiated (Log Page Identifier 07h)` talks about:

```
If the MCDAS bit (refer to Figure 215) is set to ‘1’ and a Get Log Page command with the Create Telemetry
Host-Initiated Data bit is set to ‘1’, then the maximum data area to be created in the Telemetry Host-Initiated
log page shall be less than or equal to the MCDA field in the Log Specific Parameter field in Command
Dword 10 (refer to Figure 213).
```

Is my understanding correct?

**Comment by igaw** [comment]
looks good, just a bit confused how this api works.

File: src/nvme/ioctl.h (hunk 1 above)
Comment (sc108-lee) [reply src/nvme/ioctl.h:1567]: yes it is

**Comment by sc108-lee** [comment]
Actually I was considering api problem, same as you.
So I updated the commit as I first modified.

remain : old function to call new function
new : nvme_get_log_create_telemetry_host_mcda include mcda argument.

**Comment by igaw** [comment]
Thanks!

**Pull request merged**